### ISSUE TITLE
Report that myjobs is requiring account string.

### ISSUE THREAD

**Issue by brianmcmichael** [OSC/ood-myjobs#163]
```
Hello WIAG, 
 
When I use "My Jobs" to submit jobs to all clusters, the "account" under "Job options" is mandatory instead of optional. If I do not specify "Account", I'll get the following error (this error from Owens submission):
 
A PBS::Error occurred when submitting jobs for simulation 16: ERROR: Your job has not been submitted to Owens for the following reason:
 
You may not use account /tmp/qsub.20170301-94971-b6txxk. Please choose one of the
following instead.
Available accounts: PZS0200
 
Please review the Owens documentation:
 
https://www.osc.edu/supercomputing/computing/owens
 
Feel free to contact oschelp@osc.edu if you have any questions.
 
qsub: Your job has been administratively rejected by the queueing system.
qsub: There may be a more detailed explanation prior to this notice.
 
This issue has been reported by at least two users up to now. 
```

**Comment by brianmcmichael** [comment]
Account string options were added recently. Looking through the code that was added, I don't see that any provision was given for setting a default.

**Comment by ericfranz** [comment]
If you copy an existing job and submit it, it works fine.

If you create a new job from default template and submit it, it works fine.

However, if you create a new job from default template, edit job options, and change the cluster from oakley to owens, and then submit, you'll see the error:

```
You may not use account /tmp/qsub.20170301-46371-vne698.  Please choose one of the
	following instead.
	
	Available accounts:  PAN0014
```

I think what happens is when we save job options, the Account String field is left blank and then gets set as an empty string "". This is the offending code:

https://github.com/OSC/ood-myjobs/blob/5d8326ffc12c2d03a4af1c35c01bfae83098889e/app/controllers/workflows_controller.rb#L69-L82

The account gets stored as "" instead of being omitted.

Then this is the code that mistakenly sets the account to an empty string.

https://github.com/OSC/ood-myjobs/blob/5d8326ffc12c2d03a4af1c35c01bfae83098889e/app/models/workflow.rb#L113:

```ruby
# in Workflow

  def build_jobs(staged_dir, job_list = [])
    job_list << OSC::Machete::Job.new(
      script: staged_dir.join(staged_script_name),
      host: batch_host,
      account_string: account,
      torque_helper: ResourceMgrAdapter.new
    )
  end
```

```ruby
# in ResourceMgrAdapter
  def qsub(script_path, host: nil, depends_on: {}, account_string: nil)
    ...
    script = OodJob::Script.new(content: script_path.read, accounting_id: account_string)
    ...
  end
```

 
Proper solution is probably to fix it here:

```diff
# in Workflow

  def build_jobs(staged_dir, job_list = [])
    job_list << OSC::Machete::Job.new(
      script: staged_dir.join(staged_script_name),
      host: batch_host,
-      account_string: account,
+      account_string: account.blank? ? nil : account,
      torque_helper: ResourceMgrAdapter.new
    )
  end
```

**Comment by ericfranz** [comment]
Steps to reproduce, using an account that only has 1 valid project to bill against:

1. Click "create a new job from default template"
2. Select new job and edit Job Options
3. Change the cluster from Oakley to Owens and submit the form. Leave Account blank.
4. Select and submit the job

You should see the error.

**Comment by brianmcmichael** [comment]
I feel like this could happen again if `account` is used elsewhere, I propose instead a new reader method to consistently return `nil` if the string is empty:

```
  def account
    job_attrs[:account] unless job_attrs[:account].blank?
  end
```

**Comment by brianmcmichael** [comment]
fix released in v2.1.1
